index.js

```diff
@@ -0,0 +1,61 @@
+let text = document.getElementById("text");
+let list = document.getElementById("list");
+let search = document.getElementById("search");
+text.value = "";
+search.value = "";
+let index = 0;
+
+text.addEventListener("keypress", function (e) {
+  if (e.key === "Enter") {
+    btnSubmit();
+  }
+});
+
+search.addEventListener("input", function (f) {
+  if (f.target.value != "") {
+    const searchData = data.filter((a) => a.text.includes(f.target.value));
+    displayData(searchData);
+  } else {
+    displayData(data);
+  }
+});
+
+let data = [];
+
+function btnSubmit() {
+  let txtValue = text.value;
+  if (txtValue) {
+    let task = {
+      index: index++,
+      isChecked: false,
+      text: txtValue,
+    };
+    data.push(task);
+  }
+  displayData(data);
+}
+
+function displayData(data) {
+  list.innerHTML = "";
+  if (data) {
+    for (let item of data) {
+      list.innerHTML += `<div class="checklist"><input type="checkbox" onclick="itemChecked(${data.indexOf(
+        item
+      )})" ${item.isChecked ? "checked" : ""}> ${item.text} </input></div>`;
+    }
+    text.value = "";
+  }
+}
+function itemChecked(index) {
```

Comment:
when checked the item should get strikeout instead of removed immediately.

add different button for removing items from the list.